tensor/stats/metric/funcs.go

```diff
@@ -0,0 +1,329 @@
+// Copyright (c) 2024, Cogent Core. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package metric
+
+import (
+	"math"
+
+	"cogentcore.org/core/math32"
+	"cogentcore.org/core/tensor"
+	"cogentcore.org/core/tensor/stats/stats"
+)
+
+// MetricFunc is the function signature for a metric function,
+// where the output has the same shape as the inputs but with
+// the outermost row dimension size of 1, and contains
+// the metric value(s) for the "cells" in higher-dimensional tensors,
+// and a single scalar value for a 1D input tensor.
+// Critically, the metric is always computed over the outer row dimension,
+// so each cell in a higher-dimensional output reflects the _row-wise_
+// metric for that cell across the different rows.  To compute a metric
+// on the [tensor.SubSpace] cells themselves, must call on a
+// [tensor.New1DViewOf] the sub space.  See [simat] package.
+// All metric functions skip over NaN's, as a missing value,
+// and use the min of the length of the two tensors.
+// Metric functions cannot be computed in parallel,
+// e.g., using VectorizeThreaded or GPU, due to shared writing
+// to the same output values.  Special implementations are required
+// if that is needed.
+type MetricFunc func(a, b, out *tensor.Indexed)
+
+// SumSquaresScaleFuncOut64 computes the sum of squares differences between tensor values,
+// returning scale and ss factors aggregated separately for better numerical stability, per BLAS.
+func SumSquaresScaleFuncOut64(a, b, out *tensor.Indexed) (scale64, ss64 *tensor.Indexed) {
```

Comment:
Every one of these needs a `Func` suffix?

Reply:
removed Func in Out64 cases; definitely do call these "Func"s from other packages; cannot export (also MinFunc would conflict with 'min').  I don't see a good alternative, except maybe adding a prefix to the enum that we then strip using enum stripping?  StatMean, StatSum etc?  Maybe that is better?

Reply:
fixed, enum has Stats* etc.